python/ray/serve/_private/replica.py

```diff
@@ -1554,6 +1602,11 @@ async def _call_user_record_routing_stats(self) -> Dict[str, Any]:
         result, _ = await self._call_func_or_gen(self._user_record_routing_stats)
         return result
 
+    @_run_user_code
+    async def _call_user_autoscaling_stats(self) -> Dict[str, Any]:
```

Comment:
also we should strictly validate the type IMO, if user returns anything else we should log error and return empty dictionary. The reason why string types are important is because the downstream code that does metrics aggregation will not work with return type of `Dict[str, str]`

```suggestion
    async def _call_user_autoscaling_stats(self) -> Dict[str, Union[int, float]]:
```

Reply:
I added a validation check which runs on the first call to `_add_autoscaling_metrics_point_async`, and disables `_user_autoscaling_stats_available` if it fails validation.